assemblers/http_parser_test.go

```diff
@@ -30,16 +33,20 @@ func TestExtractHeader(t *testing.T) {
 				"Host":       []string{"test"},
 				"Cookie":     []string{"test"},
 				"User-Agent": []string{"test"},
+				"X-Test":     []string{"test"},
```

Comment:
It might be useful to add a test case demonstrating the case-sensitive nature of our function, not so much for testing that it's case-sensitive, but more for "here's how it works today". Like `assert.NotEqual(X-MyHeader,x-myheader)`

Reply:
Sure, happy to add another test case.